src/opensles/AudioStreamOpenSLES.cpp

```diff
@@ -163,6 +163,45 @@ PerformanceMode AudioStreamOpenSLES::convertPerformanceMode(SLuint32 openslMode)
     return oboeMode;
 }
 
+void AudioStreamOpenSLES::logUnsupportedAttributes() {
+    // Log unsupported attributes
+    // only report if changed from the default
+
+    // Device ID
+    if (mDeviceId != kUnspecified) {
+        LOGW("Device ID [AudioStreamBuilder::setDeviceId()] "
+             "is not supported on OpenSLES streams.");
+    }
+    // Sharing Mode
+    if (mSharingMode != SharingMode::Shared) {
+        LOGW("SharingMode [AudioStreamBuilder::setSharingMode()] "
+             "is not supported on OpenSLES streams.");
+    }
+    // Performance Mode
+    int sdkVersion = getSdkVersion();
+    if (mPerformanceMode != PerformanceMode::None && sdkVersion < __ANDROID_API_N_MR1__) {
+        LOGW("PerformanceMode [AudioStreamBuilder::setPerformanceMode()] "
+             "is not supported on OpenSLES streams running on pre-Android N-MR1 versions.");
+    }
+    // Content Type
+    if (mContentType != ContentType::Music) {
+        LOGW("ContentType [AudioStreamBuilder::setContentType()] "
+             "is not supported on OpenSLES streams.");
+    }
+
+    // Session Id
+    if (mSessionId != SessionId::None) {
+        LOGW("SessionId [AudioStreamBuilder::setSessionId()] "
+             "is not supported on OpenSLES streams.");
+    }
+
+    // Input Preset
+    if (mInputPreset != InputPreset::VoiceRecognition) {
+        LOGW("InputPreset [AudioStreamBuilder::setInputPreset()] "
+             "is not supported on OpenSLES streams.");
+    }
+}
```

Comment:
Isn't this wrong? It looks like input presets on OpenSL ES have been supported since commit c30bbe603c256f92cdf2876c3122bc5be24b5e3e, which landed well before this one.

Reply:
Thanks. I creates Issue #1043 for this.